docs/proposal/20211210-launch-templates-managedmachinepools.md

```diff
@@ -0,0 +1,254 @@
+    * AMIVersion
+    * AMIType
+    * DiskSize
+    * InstanceType
+3. Add new `LaunchTemplateID` and `LaunchTemplateVersion` fields to [AWSManagedMachinePoolStatus](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/9bc29570614aa7123d79f042b6e6efc2aaf3e490/exp/api/v1beta1/awsmanagedmachinepool_types.go#L171) to store details of the launch template and version used.
+4. Add a new `LaunchTemplateVersion` field to [AWSMachinePoolStatus](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/api/v1beta1/awsmachinepool_types.go#L112) to store the version of the launch template used.
+5. [Refactor the code](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/ec057ad6e613a6578f67bf68a6c77fbe772af933/exp/controllers/awsmachinepool_controller.go#L383) from the `AWSMachinePool` controller that reconciles `AWSLaunchTemplate` into a common location so that it can be shared.
+6. Update the controller for `AWSManagedMachinePool` to use the `AWSLaunchTemplate` reconciliation logic.
+7. Add checks in the `AWSManagedMachinePool` create/update validation webhooks that stops users specifying `AWSLaunchTemplate` if fields `AMIType,AMIVersion,InstanceType,DiskSize,InstanceProfile` are set
+8. Add warning logs to the `AWSManagedMachinePool` create/update validation webhooks if fields `AMIType,AMIVersion,InstanceType,DiskSize,InstanceProfile` stating that these fields will be deprecated in the future and that `AWSLaunchTemplate` should be used instead
+> An area that is undecided upon is should we auto convert the `AMIType,AMIVersion,InstanceType,DiskSize,InstanceProfile` fields if specified into a `AWSLaunchTemplate`. We should investigate this as part of implementation. 
+10. Update the cluster templates that use `AWSManagedMachinePool` so that they use `AWSLaunchTemplate`
+11. Update the API version roundtrip tests for v1alpha4<->v1beta1 conversions of `AWSManagedMachinePool`
+12. Update the EKS e2e tests to add an additional test step where we create an additional managed machine pool using  `AWSLaunchTemplate`.
+13. Update any relevant documentation
+14. Release note must mention that "action is required" in the future, as fields are being deprecated.
+15. As part of the implementation, we should investigate how in the f
+
+### User Stories
+
+#### Story 1
+
+AS a CAPA user
+I want to create a managed machine pool using a launch template
+So that I can use functionality from the AWS launch template
+
+#### Story 2
+
+As a CAPA user
+I want to have consistency between managed and unmanaged machine pools
+So that I can choose which to use based on whether I want managed and not based on missing functionality
+
+#### Story 3
+
+As a CAPA user
+I want to ensure launch templates are versioned
+So that I can see a history of the changes in the console
```

Comment:
how do we see the version history?

Reply:
The version history is seen in the AWS console for the launch template. 

We could also add a command to `clusterawsadm` to list them.....do you think that would be helpful?